### DISCUSSION TITLE
Can I use this library to convert Gerber files into 3D models?

### DISCUSSION THREAD

**Discussion by chenzanyu** [Argmaster/pygerber#315] [General]
It's known that SVG images can be generated. How can they be converted into 3D models? Do you have any ideas about this? Thank you very much. @Argmaster

**Comment by Argmaster** [comment]
Hi, so SVGs created with shapely backend (currently on main, will be available in 3.0.0a3) are kind of suitable for that. You can import those SVGs into Blender (either with GUI or with bpy module available on pypi) as grease pencil and convert that into mesh and fill with polygons, then you can use solidify modifier to add thickness. Unfortunately this is not trivial to automate as Gerber images can have planes with holes which do not connect to any of the edges. This kind of topology doesn't work well with Blender SVG importing. I am still working on that, but there is no simple solution. 

SVG renderer available in 2.4.1 is not suitable for 3D model generation as it does not perform actual difference (cutting) operations, it only masks regions which have to be transparent, which does not yield satisfying results in Blender.

**Reply by chenzanyu** [reply]
Thank you very much for your reply. It seems that I only need to wait for the new version to be released to proceed with the verification.

**Comment by Argmaster** [comment]
Hi @chenzanyu

I have published [PyGerber 3.0.0a3](https://github.com/Argmaster/pygerber/releases/tag/v3.0.0a3), it is still a alpha release, but for image generation it should be mature enough. 

You can install it from PyPI with:

```shell
pip install pygerber==3.0.0a3
```

You can find quick start guide [here](https://argmaster.github.io/pygerber/latest/70_gerber/20_quick_start/01_single_file.html) in particular SVG image generation guide in [Vector images](https://argmaster.github.io/pygerber/latest/70_gerber/20_quick_start/01_single_file.html#vector-images) section.

**Comment by chenzanyu** [comment]
```
from pygerber.gerber.api import GerberFile, Style

from pygerber.examples import ExamplesEnum, get_example_path

path_to_gerber_file = get_example_path(ExamplesEnum.UCAMCO_2_11_2)

image = GerberFile.from_file(path_to_gerber_file).render_with_shapely(
    Style.presets.BLACK_WHITE
)
image.save("output.svg")
```

I installed the latest 3.0.0a3 version, and I cannot preview the SVG image generated by the example code in the browser.

**Reply by Argmaster** [reply]
Hi, is the image generated at all? I forgot to mention that vector rendering requires shapely to be installed, make sure you have it:

```
pip install pygerber[shapely]==3.0.0a3
```

or 

```
pip install shapely
```

If the issue appears despite you having shapely installed, please [open an issue with vm issue template](https://github.com/Argmaster/pygerber/issues/new?assignees=Argmaster&labels=virtual-machine%2Cbug%2Cwaiting-for-checkboxes&projects=&template=pygerber_vm_bug.md&title=%5BPyGerber%5D%3A+%3Cwhen-i-am-doing-something-then-something-fails%3E)

**Reply by chenzanyu** [reply]
Hi @Argmaster   Using pygerber Version a3 to Determine Units of a Gerber File？

**Reply by chenzanyu** [reply]
o, I found it
